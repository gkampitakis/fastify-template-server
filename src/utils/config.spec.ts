describe('Configuration', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  describe('Production config', () => {
    it('Default config', async () => {
      process.env.NODE_ENV = 'production';
      const { default: config } = await import('./config');

      expect(config).toEqual({
        server: {
          port: 4000
        },
        isProduction: true,
        cors: {
          origin: false
        },
        healthCheck: {
          exposeFailure: false,
          info: { Service: 'Template' },
          path: '/api/health'
        },
        shutdownDelay: 5000
      });
    });

    it('Logger', async () => {
      process.env.NODE_ENV = 'production';

      const { logger } = await import('./config');

      expect(logger).toEqual({
        level: 'warn',
        base: {
          name: 'Template'
        },
        enabled: true,
        prettyPrint: false
      });
    });
  });

  describe('Development config', () => {
    it('Default config', async () => {
      process.env.NODE_ENV = 'dev';
      const { default: config } = await import('./config');

      expect(config).toEqual({
        server: {
          port: 4000
        },
        isProduction: false,
        cors: {
          origin: false
        },
        healthCheck: {
          exposeFailure: true,
          info: { Service: 'Template' },
          path: '/api/health'
        },
        shutdownDelay: 5000
      });
    });

    it('Logger', async () => {
      process.env.NODE_ENV = 'dev';

      const { logger } = await import('./config');

      expect(logger).toEqual({
        level: 'debug',
        base: {
          name: 'Template'
        },
        enabled: true,
        prettyPrint: {
          colorize: true,
          ignore: 'hostname,pid',
          translateTime: 'UTC:yyyy-mm-dd\'T\'HH:MM:ss',
          levelFirst: true
        }
      });
    });
  });
});