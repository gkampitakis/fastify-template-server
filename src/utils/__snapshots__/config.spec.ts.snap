// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Development config Default config 1`] = `
Object {
  "cors": Object {
    "origin": false,
  },
  "healthCheck": Object {
    "exposeFailure": true,
    "info": Object {
      "Service": "test service",
    },
    "path": "/api/health",
  },
  "isProduction": false,
  "server": Object {
    "port": 4000,
  },
  "shutdownDelay": 5000,
}
`;

exports[`Configuration Development config Logger 1`] = `
Object {
  "base": Object {
    "name": "test service",
  },
  "enabled": true,
  "level": "debug",
  "transport": Object {
    "options": Object {
      "colorize": true,
      "ignore": "hostname,pid",
      "levelFirst": true,
      "translateTime": "UTC:yyyy-mm-dd'T'HH:MM:ss",
    },
    "target": "pino-pretty",
  },
}
`;

exports[`Configuration Production config Default config 1`] = `
Object {
  "cors": Object {
    "origin": false,
  },
  "healthCheck": Object {
    "exposeFailure": false,
    "info": Object {
      "Service": "",
    },
    "path": "/api/health",
  },
  "isProduction": true,
  "server": Object {
    "port": 4000,
  },
  "shutdownDelay": 5000,
}
`;

exports[`Configuration Production config Logger 1`] = `
Object {
  "base": Object {
    "name": "test service",
  },
  "enabled": true,
  "level": "info",
}
`;
